src/bytes.rs

```diff
@@ -250,19 +395,30 @@ impl Ord for Bytes<'_> {
         self.as_slice().cmp(other)
     }
 }
-impl<'a> From<&'a [u8]> for Bytes<'a> {
-    fn from(data: &'a [u8]) -> Self {
-        static VT: BytesVt = BytesVt {
-            release: None,
-            retain: Some(noop),
-        };
-        Bytes {
-            start: data.as_ptr().cast(),
-            len: data.len(),
-            data: data.as_ptr().cast(),
-            capacity: data.len(),
-            vtable: &VT,
-        }
+
+#[cfg(feature = "std")]
+impl std::io::Read for Bytes<'_> {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        let len = self.len().min(buf.len());
```

Comment:
We'll probably end up adding a lint against postfix `.min()` and `.max()`, which can be mistaken with the opposite `.clamp()`ing operations. In that regard, either use the prefix notation:

```suggestion
        let len = usize::min(self.len(), buf.len());
```

or the `.clamp()`ing operation:

```suggestion
        let len = self.len().clamp(0, buf.len());
```

(here I'd favor the first one since the `copy_from_slice` is a very symmetrical operation)